### PR TITLE
Created test suite and Added flat discount test

### PR DESCRIPTION
**Pull request by Anthony-dominianni**
Made some tests with ben noob

### CODE REVIEW

File: src/main/java/edu/neu/cs4500/models/Estimate.java [hunk 1]
```diff
@@ -24,6 +25,39 @@
     
     @OneToMany(mappedBy="estimate")
     private List<Fee> fees;
+    
+    @OneToMany(mappedBy="estimate")
+    private List<Discount> discounts;
+    
+    public float getDiscount() {
+//    	List<Discount> subscriptionDiscounts = discounts.stream()
```
Comment (benjaminclauss) [inline src/main/java/edu/neu/cs4500/models/Estimate.java]: Remove this beautiful clean code.

File: src/main/java/edu/neu/cs4500/models/Estimate.java [hunk 2]
```diff
@@ -24,6 +25,39 @@
...
+//    			.collect(Collectors.toList());
+//    	
+//    	List<Discount> flatDiscounts = discounts.stream()
+//    			.filter(discount -> discount.isFlat())
+//    			.collect(Collectors.toList());
+    	
+    	
+    	float totalDiscount = 0;
+    
+    	for (Discount discount : discounts) {
```
Comment (benjaminclauss) [inline src/main/java/edu/neu/cs4500/models/Estimate.java]: https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html

File: src/main/java/edu/neu/cs4500/models/Estimate.java [hunk 3]
```diff
@@ -24,6 +25,39 @@
...
+    			totalDiscount += discount.getDiscount();
+    		} else {
+    			if (this.subscriptionFrequency.equals(discount.getFrequency())) {
+    				totalDiscount += (this.basePrice * discount.getDiscount());
+    			}
+    			
+    		}
+    	}
+    	
+    	if (totalDiscount > basePrice) {
```
Comment (benjaminclauss) [inline src/main/java/edu/neu/cs4500/models/Estimate.java]: semantics!!!

File: src/test/java/edu/neu/cs4500/models/SubscriptionEstimateDiscountTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	private static Estimate e1;
+	private static Estimate e2;
+	private static Estimate e3;
+	private static Discount d1;
+	private static Discount d2;
+	private static Discount flat;
+	
+	@BeforeAll
+    public static void setUp() {
```
Comment (benjaminclauss) [inline src/test/java/edu/neu/cs4500/models/SubscriptionEstimateDiscountTest.java]: Break up this set-up logic properly.

**Comment by ysmolyar** [comment]
ok nvm i'm noob just do what master ben says

**Comment by benjaminclauss** [comment]
image

**Comment by ysmolyar** [comment]
F

**Pull request merged**